src/Features/LanguageServer/Protocol/Handler/BufferedProgress.cs

```diff
@@ -13,21 +15,21 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
     /// expected to be.  Namely, multiple client can be calling <see cref="IProgress{T}.Report(T)"/> on it at the same
     /// time.  This is safe, though the order that the items are reported in when called concurrently is not specified.
     /// </summary>
-    internal readonly struct BufferedProgress<T> : IProgress<T[]>, IProgress<T>, IDisposable
+    internal readonly struct BufferedProgress<T> : IProgress<T>, IDisposable
```

Comment:
IProgress<T> and IProgress<T[]> was confusing me and causing issues with the new protocol types (the diagnostic reports are single objects, not arrays of objects).

I removed IProgress<T[]> so that this always cleanly can represent any kind of T (either reporting an object or an array), plus added a couple helpers for reporting arrays.

Reply:
hrmm.  i'm on the fence.  but up to you :)